lib/db/mongodb/errors.go

```diff
@@ -9,14 +9,18 @@ const (
 )
 
 func HasErrorCode(err error, code int) bool {
-	we, ok := err.(mongo.WriteException)
-	if !ok {
-		return false
-	}
-
-	for _, err := range we.WriteErrors {
-		if err.Code == code {
-			return true
+	switch v := err.(type) {
+	case mongo.WriteException:
+		for _, err := range v.WriteErrors {
```

Comment:
Could set `v.WriteErrors` on each case to a variable outside the switch, then you can avoid any duplication of the looping/code checking

Reply:
But then if the error is not for write or bulk write it would break?

Reply:
Mea culpa, Mongo has only errors for `WriteException` and `BulkWriteException` 🧐

Reply:
Addressed